crates/shared/src/price_estimation/trade_verifier/mod.rs

```diff
@@ -167,23 +177,20 @@ impl TradeVerifier {
             )
             .tx;
 
-        let simulation = solver
-            .methods()
-            .swap(
-                self.settlement.address(),
-                tokens.clone(),
-                verification.receiver,
-                Bytes(settlement.data.unwrap().0),
-            )
-            .tx;
+        let swap_simulation = Solver::Solver::swapCall {
+            settlementContract: self.settlement.address().into_alloy(),
+            tokens: tokens.iter().cloned().map(IntoAlloy::into_alloy).collect(),
+            receiver: verification.receiver.into_alloy(),
+            settlementCall: alloy::primitives::Bytes::from(settlement.data.unwrap().0),
+        };
 
         let block = *self.block_stream.borrow();
 
         let call = CallRequest {
```

Comment:
Theoretically we could simplify this call and decoding the result significantly with `alloy` by just calling `SolverInstance::swap()` together with the states overrides.
However, at the moment the code that causes tenderly simulation commands to be logged for every quote verification lives inside the `Arc<dyn CodeSimulating>` thingy.
So if we move the tenderly logging logic out of that and perhaps just have a helper function that prints the corresponding tenderly log we could clean this up very nicely.

Reply:
This is much more involved than it seems 😅 moving this to an issue, merging the current PR and handling the issue before picking up anything else